### PR TITLE
Remove the sh dependency

### PR DESCRIPTION
**Pull request by encukou**
This converts from `sh` to `subprocess`.

### CODE REVIEW

File: elsa/_deployment.py [hunk 1]
```diff
@@ -10,29 +10,39 @@
...
 ]
 
+def run(cmd, **kwargs):
+    """Same as ``subprocess.run``, but checks the result by default"""
+    kwargs.setdefault('check', True)
+    return subprocess.run(cmd, **kwargs)
+
+
+def get_last_commit_info(format):
+    cmd = ['git', 'show', '--format', format, '--no-patch']
```
Comment (hroncok) [inline elsa/_deployment.py]: Consider adding `--no-pager` just after `git`. The pager tends to make trouble on some environments and only sometimes.

Comment (encukou) [inline elsa/_deployment.py]: Added.

**Comment by hroncok** [comment]
Also, I'm not sure it actually fixes https://github.com/pyvec/elsa/issues/16 just yet. Did you test it on Windows? If not, please change it to `Releated` or something like that.

**Comment by encukou** [comment]
#16 is just for `sh`. I filed #20 for testing on Windows.

**Comment by hroncok** [comment]
OK.

**Comment by hroncok** [comment]
I'll squash, as I don't think the extra commit are worth keeping separately, is that OK?

**Comment by encukou** [comment]
Sure.

**Pull request merged**